packages/react-devtools-shared/src/__tests__/__snapshots__/profilingCache-test.js.snap

```diff
@@ -3207,6 +3244,2081 @@ Object {
+  "updaters": Array [
+    Object {
+      "displayName": "Anonymous",
+      "hocDisplayNames": null,
+      "id": 1,
+      "key": null,
+      "type": 11,
+    },
+  ],
+}
+`;
+
+exports[`ProfilingCache should properly detect when a component is missing a shallow/deep memo for props: CommitDetails commitIndex: 1 1`] = `
+Object {
+  "changeDescriptions": null,
+  "duration": 0,
+  "effectDuration": null,
+  "fiberActualDurations": Map {
+    2 => 0,
+    1 => 0,
+  },
+  "fiberSelfDurations": Map {
+    2 => 0,
+    1 => 0,
+  },
+  "passiveEffectDuration": null,
+  "perfInsights": Map {
+    2 => Object {
+      "arePropsDeepEqual": true,
+      "arePropsEqual": true,
+      "didHooksChange": false,
+      "isContextDeepEqual": false,
+      "isContextEqual": true,
+      "isFirstMount": false,
+      "isStateDeepEqual": true,
+      "isStateEqual": true,
+      "nonMemoizedProps": null,
+    },
+  },
+  "priorityLevel": "Immediate",
+  "timestamp": 0,
+  "updaters": Array [
+    Object {
+      "displayName": "Anonymous",
+      "hocDisplayNames": null,
+      "id": 1,
+      "key": null,
+      "type": 11,
+    },
+  ],
+}
+`;
+
+exports[`ProfilingCache should properly detect when a component is missing a shallow/deep memo for props: CommitDetails commitIndex: 2 1`] = `
+Object {
+  "changeDescriptions": null,
+  "duration": 0,
+  "effectDuration": null,
+  "fiberActualDurations": Map {
+    2 => 0,
+    3 => 0,
+    1 => 0,
+  },
+  "fiberSelfDurations": Map {
+    2 => 0,
+    3 => 0,
+    1 => 0,
+  },
+  "passiveEffectDuration": null,
+  "perfInsights": Map {
+    2 => Object {
+      "arePropsDeepEqual": false,
+      "arePropsEqual": false,
+      "didHooksChange": false,
+      "isContextDeepEqual": false,
+      "isContextEqual": true,
+      "isFirstMount": false,
+      "isStateDeepEqual": true,
+      "isStateEqual": true,
+      "nonMemoizedProps": null,
+    },
+    3 => Object {
+      "arePropsDeepEqual": false,
+      "arePropsEqual": false,
+      "didHooksChange": false,
+      "isContextDeepEqual": false,
+      "isContextEqual": true,
+      "isFirstMount": false,
+      "isStateDeepEqual": true,
+      "isStateEqual": true,
+      "nonMemoizedProps": null,
+    },
+  },
+  "priorityLevel": "Immediate",
+  "timestamp": 0,
+  "updaters": Array [
+    Object {
+      "displayName": "Anonymous",
+      "hocDisplayNames": null,
+      "id": 1,
+      "key": null,
+      "type": 11,
+    },
+  ],
+}
+`;
+
+exports[`ProfilingCache should properly detect when a component is missing a shallow/deep memo for props: CommitDetails commitIndex: 3 1`] = `
+Object {
+  "changeDescriptions": null,
+  "duration": 0,
+  "effectDuration": null,
+  "fiberActualDurations": Map {
+    2 => 0,
+    4 => 0,
+    1 => 0,
+  },
+  "fiberSelfDurations": Map {
+    2 => 0,
+    4 => 0,
+    1 => 0,
+  },
+  "passiveEffectDuration": null,
+  "perfInsights": Map {
+    2 => Object {
+      "arePropsDeepEqual": true,
+      "arePropsEqual": true,
```

Comment:
I think this is wrong? Props shallowly changed for both components that rendered this time, no?

Reply:
```
    // Fourth render has deep changes only.
    utils.act(() =>
      legacyRender(
        <>
          <ComponentNeedsMemo count={2} />
          <ComponentShallowMemo count={2} />
          <ComponentNeedsDeepMemo style={{color: 'red'}} />
          <ComponentDeepMemo style={{color: 'red'}} />
        </>,
        container,
      ),
    );
```

In this case, `ComponentNeedsMemo` doesn't have a `React.memo` so its correctly reporting that its props are equal since previous render.

Reply:
Ah, I thought this was for the previous render (when props shallowly changed). This is the problem with external snapshots :)